sdks/python/setup.py

```diff
@@ -17,29 +17,42 @@
 
 """Apache Beam SDK setup configuration."""
 
-import os
 import platform
-import re
 import setuptools
+import re
+from apache_beam.version import get_version_from_pom
 
 
-# Currently all compiled modules are optional  (for performance only).
+# Currently all compiled modules are optional (for performance only).
 if platform.system() == 'Windows':
-  # Windows doesn't always provide int64_t.
-  cythonize = lambda *args, **kwargs: []
-else:
-  try:
-    # pylint: disable=g-statement-before-imports,g-import-not-at-top
-    from Cython.Build import cythonize
-  except ImportError:
+    # Windows doesn't always provide int64_t.
     cythonize = lambda *args, **kwargs: []
+else:
+    try:
+        # pylint: disable=g-statement-before-imports,g-import-not-at-top
+        from Cython.Build import cythonize
+    except ImportError:
+        cythonize = lambda *args, **kwargs: []
 
 
-def get_version():
-  global_names = {}
-  execfile(os.path.normpath('./apache_beam/version.py'),
-           global_names)
-  return global_names['__version__']
+# Reads the actual version from pom.xml file, and synchronizes
+# apache_beam.__version__ field for later usage.
+def sync_version():
```

Comment:
I do not think you need this if the get_version_from_distribution() really works. you can use get_version_from_pom in setup.py and use get_version_from_distribution() in all the other places. This way you avoid the somewhat fragile rewriting the __init_.py file.